app/frontend/actions.ts

```diff
@@ -220,21 +220,17 @@ export default ({setState, getState}: {setState: SetStateFn; getState: GetStateF
     })
   }
 
-  const updateMnemonic = async (state, e) => {
+  const updateMnemonic = (state, e) => {
     const mnemonicInputValue = e.target.value
     setState({
       mnemonicInputValue,
-      showMnemonicValidationError: false,
+      mnemonicValidationError: null,
     })
-    handleError(
-      'mnemonicValidationError',
-      await mnemonicValidator(sanitizeMnemonic(mnemonicInputValue))
-    )
```

Comment:
@PeterBenc implemented handleError function for unifying error handling, which sets the state for specified error text. Nearly all errors are handled this way. Did you find it this part of code messy or why change it?

Reply:
I am strongly against mis-using "handleError" as means of validation, given its name it should handle a real errors (e.g., failed to submit tx, internal error, etc.). [I guess we can change the name though. However, I am not happy it puts everything into top-level state.]
Speaking of which, `handleError` is terrible name for something which does not handle anything, it should be `setError` in the first place

Reply:
I defentlly agree with the `setError` name, and I agree putting everything into top level state isnt good idea either, the purpose of the `handleError` was to save some duplicate lines. The situation was that we were putting almost all errors into top level state, the workflow was always the same, set error state and then have it checked whether it's unexpected thus sending it to sentry. So instead of doing this in all cases, you could just call handleError with name and value that should be set.

As I mentioned, `setError` is a lot more appropriate and when it comes to "puts everything into to level state", we can change that and then the `handleError` function wont be nessesary and we can find a different way to always check for unexpected errors.

Reply:
Let's gradually figure out the final design. For now let's move relevant validation "errors" into corresponding substates and handle them in their respective actions. But I would do this only when touching given part of codebase ...